### PR TITLE
Added Aimsun tools to the XTMF GUI.

### PR DESCRIPTION
**Pull request by amitsandhel**
Created new .cs files inside the TMG.Aimsun folder.
Created the AimsunTool and the IAimsunTool files
Cleaned up code base removed dead code
Note that only the importNetwork test module has  dead code removed for this PR request,
ToDo: The other test scripts will be refactored when the other aimsun tool will be properly intergrated into XTMF
Currently only ImportNetwork is properly integrated

### CODE REVIEW

File: aimsunBridge.py [hunk 1]
```diff
@@ -169,7 +151,9 @@ def executeAimsunScript(self, moduleDict, console, model):
         """
         try:
             # we need to append the Toolbox/InputPut folder path so all relative imports will work
-            sys.path.append(moduleDict["parameters"]["ToolboxInputOutputPath"])
+            toolDirectory = os.path.dirname( moduleDict["toolPath"])
+            print ('tool directory ', toolDirectory)
```
Comment (JamesVaughan) [inline aimsunBridge.py]: Cleanup debugging code + fix bracket.

File: src/TMG.Aimsun/AimsunController.cs [hunk 2]
```diff
@@ -93,6 +94,11 @@ public sealed class ModellerController : IDisposable
         /// </summary>
         private const int SignalSentPrintMessage = 11;
 
+        internal bool Run(ImportNetwork importNetwork, string toolName, AimsunControllerParameter[] aimsunControllerParameters)
+        {
+            throw new NotImplementedException();
+        }
+
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/AimsunController.cs]: This looks like some accidental autogenerated code.

File: src/TMG.Aimsun/AimsunControllerParameter.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        internal string Name;
+        internal string Value;
+        public AimsunControllerParameter(string name, string value)
+        {
+            Name = name;
+            Value = value;
+        }
+    }
+}
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/AimsunControllerParameter.cs]: Is this class still used?  If not please remove.

File: src/TMG.Aimsun/AimsunTool.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using XTMF;
+
+namespace TMG.Aimsun
+{
+    [ModuleInformation(Description = @"The aimsun tool to run it takes an aimsun and a list of all the arguments you 
+    need to run it ")]
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/AimsunTool.cs:26]: With the @ string, all of spaces on line 7 will also be copied to the documentation.  You might want to align that on the left or just concatenate the string by parts.  The compiler will automatically add it all together.

File: src/TMG.Aimsun/AimsunTool.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    public class AimSunTool : IAimsunTool
+    {
+        [RunParameter("Tool Arguments", "", "The arguments of the Aimsun tool you want to run")]
+        public string ToolArguments;
+
+        [RunParameter("Tool Name", "", "The namespace of the Emme tool you want to run")]
+        public string ToolName;
+        public string Name
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/AimsunTool.cs]: Missing a line break.

File: src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+namespace TMG.Aimsun
+{
+    public class ExecuteToolsFromModellerResource: ISelfContainedModule
+    {
+        [SubModelInformation(Required = false, Description = "The tools to run in order.")]
+        public IAimsunTool[] Tools;
+
+        //use i datasource of aimsunmodeller here and this would call loadsimsuncontroller
+        //little bit nastier this way not know the type and use this lookup 
+        //dummy resource with aimsun controller getting a void pointer
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs]: Remove personal notes.

File: src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using XTMF;
+
+namespace TMG.Aimsun
+{
+    public class ExecuteToolsFromModellerResource: ISelfContainedModule
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs:26]: Include ModuleInformation attribute for module class.

File: src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+            _Progress = () => 0f;
+        }
+
+        public string Name
+        {
+            get;
+            set;
+        }
+        private Func<float> _Progress = () => 0f;
+        public float Progress
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs]: Missing line break.

File: src/TMG.Aimsun/LoadAimsunController.cs [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+using XTMF;
+
+namespace TMG.Aimsun
+{
+    public class LoadAimsunController : IDataSource<ModellerController>, IDisposable
+    {
+        [SubModelInformation(Required = true, Description = "The location of the aimsun project file.")]
+
+        private ModellerController Data;
+        public ModellerController GiveData() => Data;
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/LoadAimsunController.cs]: Fix line spacing.  Remove SubModelInformation attribute.

File: src/TMG.Aimsun/LoadAimsunController.cs [hunk 10]
```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using TMG.Input;
+using XTMF;
+
+namespace TMG.Aimsun
+{
+    public class LoadAimsunController : IDataSource<ModellerController>, IDisposable
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/LoadAimsunController.cs:27]: Add ModuleInformation attribute.

File: src/TMG.Aimsun/LoadAimsunController.cs [hunk 11]
```diff
@@ -0,0 +1,77 @@
+﻿using System;
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/LoadAimsunController.cs]: Add copyright header.

File: src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs [hunk 12]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs]: Add copyright header.

File: src/TMG.Aimsun/AimsunTool.cs [hunk 13]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
```
Comment (JamesVaughan) [inline src/TMG.Aimsun/AimsunTool.cs]: Add copyright header.

**Review by JamesVaughan — CHANGES_REQUESTED** [review]
In addition to the rest of the comments we should also remove the rest of the stub tool classes from this PR.

File: src/TMG.Aimsun/AimsunControllerParameter.cs (hunk 3 above)
Comment (amitsandhel) [reply src/TMG.Aimsun/AimsunControllerParameter.cs]: deleted

File: src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs (hunk 6 above)
Comment (amitsandhel) [reply src/TMG.Aimsun/ExecuteToolsFromModellerResource.cs]: removed personal notes

**Review by JamesVaughan — APPROVED** [review]

**Pull request merged**